### PR TITLE
[CN-279] Add support for development using Tilt

### PR DESCRIPTION
**Pull request by mtyazici**
Changes:
- Added `Tiltfile` and `Dockerfile.tilt` for development using Tilt. Also updated Makefile for the same purpose.

### CODE REVIEW

File: config/default/kustomization.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 resources:
-- ../crd
 - ../rbac
 - ../manager
+- ../crd
```
Comment (hakanmemisoglu) [inline config/default/kustomization.yaml:6]: @mtyazici What is the purpose of this change?

File: config/manager/remove_security_context.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+- op: remove
+  path: "/spec/template/spec/securityContext"
```
Comment (hakanmemisoglu) [inline config/manager/remove_security_context.yaml:2]:
```suggestion
  path: "/spec/template/spec/securityContext"
```
```suggestion
  path: "/spec/template/spec/securityContext"
  
```

File: .gitignore [hunk 3]
```diff
@@ -25,3 +25,6 @@ lintbin/*
 *.swp
 *.swo
 *~
+
+# Source code for external modules used in the Tiltfile
+tilt_modules/
```
Comment (hakanmemisoglu) [inline .gitignore:30]:
```suggestion
tilt_modules/
```
```suggestion
tilt_modules/

```

**Review by hakanmemisoglu — APPROVED** [review]
LGTM. I have just small suggestions and a question.

File: config/default/kustomization.yaml (hunk 1 above)
Comment (mtyazici) [reply config/default/kustomization.yaml:6]: It is for the `make undeploy-keep-crd` command. It deletes and at the end adds the crd to this file. However, when it re-adds the crd it appears at the end of the list. I changed it so that in the future we don't need to put the crd at top of the list manually.

File: Tiltfile [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+image_name='hz-operator-dev'
+
+# to allow connection to remote k8s clusters
+if os.getenv("ALLOW_REMOTE", default= "false").lower() == "true":
+  allow_k8s_contexts(k8s_context())
+
+# to allow using ttl.sh as container image registry, if your k8s cluster does not have one.
+if os.getenv("USE_TTL_REG", default= "false").lower() == "true":
+  registry_name='hpo-%s' % local('uuidgen')
+  default_registry('ttl.sh/%s' % registry_name.strip("\n"))
```
Comment (hasancelik) [inline Tiltfile:16]:
```suggestion
  default_registry('ttl.sh/%s' % registry_name.strip("\n").lower())
```

File: Tiltfile [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+local_resource(
+  'go-compile',
+  'make build-tilt',
+  deps=['./main.go','api/','controllers/',],
+  ignore=['api/v1alpha1/zz_generated.deepcopy.go'],
```
Comment (hasancelik) [inline Tiltfile:22]:
```suggestion
  ignore=['api/v1alpha1/zz_generated.deepcopy.go*'],
```

File: DEVELOPER.md [hunk 6]
```diff
@@ -20,6 +20,36 @@ To remove the operator from the Kubernetes cluster, run the following command:
...
+Tilt will not connect to remote clusters by default. If you want to use Tilt with any cluster, you can run the following command.
+
+```
+ALLOW_REMOTE=true tilt up
+```
+
+This will allow tilt to connect to cluster in its current context. However, if the cluster doesn't have a container registry, Tilt will not be able to push the operator image. You can use [ttl.sh](https://ttl.sh/) as the image registry by running the following command.
+
+```shell
+ALLOW_REMOTE=true USE_TTL_REG=true tilt up
```
Comment (hasancelik) [inline DEVELOPER.md:50]: same

File: DEVELOPER.md [hunk 7]
```diff
@@ -20,6 +20,36 @@ To remove the operator from the Kubernetes cluster, run the following command:
...
+- go-compile: If triggered, it builds the operator binary and triggers operator restart in the pod.
+- run e2e-test in current namespace: It will run `make e2e-test` in the current context's namespace. You can stop it by pressing cancel button. if there are any leftover resources, you can delete them by pressing the `Delete CRs and PVCs` in `hazelcast-platform-controller-manager` resource. 
+- uncategorized: Applies CRD and RBAC resources for the operator. Re-triggering it will delete and recreate mentioned resources. If re-triggered, operator pod needs to be restarted to work correctly.
+
+### Using tilt with Remote Clusters
+
+Tilt will not connect to remote clusters by default. If you want to use Tilt with any cluster, you can run the following command.
+
+```
+ALLOW_REMOTE=true tilt up
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply DEVELOPER.md:44]: I will do it

**Review by hasancelik — APPROVED** [review]
